woocommerce/woocommerce
# Separate PHP and JS Unit Tests

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Since some projects might have both kinds of tests we are going to have a `test:php` and a `test:js` script for each. This is a prerequisite to #40572 because we need to be able to easily identify what kind of tests a project has and execute them directly. We've also standardized the usage of `test:env:start` to spin up the test environment for similar reasons.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Since CI runs `pnpm test` on every project it should be good enough to examine it. We've aliased all of the `"test"` scripts to `"test:js"` where necessary and so all of the old and new commands will get hit. **Just make sure to look through and confirm that tests actually ran and it didn't pass with no command. The current workflow will let that happen!**
2. Passing PHPUnit tests for Core and other plugins is also a great indicator.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

This is just a change to developer commands.

</details>
